packages/cli/src/ui/App.tsx

```diff
@@ -94,6 +95,76 @@ interface AppProps {
   version: string;
 }
 
+function handleAutoUpdate(
+  info: UpdateInfo | null,
+  setUpdateInfo: (info: UpdateInfo | null) => void,
+  setIsUpdating: (isUpdating: boolean) => void,
+  addItem: (item: HistoryItem, timestamp: number) => void,
+) {
+  if (!info) {
+    return;
+  }
+
+  setUpdateInfo(info);
+  if (process.env.GEMINI_CLI_DISABLE_AUTOUPDATER === 'true') {
```

Comment:
can you put this in settings.ts / settings.json rather than adding as an env variable?
I could also see the case for making this an env variable but would default to just surfacing it in settings.ts as that should be sufficient and it is easier if you don't need to merge settings from multiple sources.
the case for why this needs to be settable via settings.json is that some orgs might want to disable this for everyone in their org and that is easiest to accomplish via the existing mechanism for org level settings.json files.